apps/admin-ui/src/spa/recurring-reservations/application-rounds/[id]/review/RejectedOccurrencesTable.tsx

```diff
@@ -0,0 +1,189 @@
+import React from "react";
+import { CustomTable, ExternalTableLink } from "@/component/Table";
+import { getApplicationUrl, getReservationUrl } from "@/common/urls";
+import type { RejectedOccurrencesQuery } from "@gql/gql-types";
+import { truncate } from "common/src/helpers";
+import { IconLinkExternal } from "hds-react";
+import { memoize } from "lodash";
+import { useTranslation } from "react-i18next";
+import type { TFunction } from "i18next";
+import { getApplicantName } from "@/helpers";
+import { convertWeekday } from "common/src/conversion";
+
+const unitsTruncateLen = 23;
+const applicantTruncateLen = 20;
+
+type QueryData = NonNullable<RejectedOccurrencesQuery["rejectedOccurrences"]>;
+type Edge = NonNullable<QueryData["edges"]>[0];
+type Node = NonNullable<NonNullable<Edge>["node"]>;
+type Props = {
+  sort: string | null;
+  sortChanged: (field: string) => void;
+  rejectedOccurrences: Node[];
+  isLoading?: boolean;
+};
+
+type RejectedOccurrencesView = {
+  key: string;
+  applicationPk?: number;
+  pk?: number;
+  applicantName?: string;
+  name: string;
+  unitName?: string;
+  allocatedReservationUnitName?: string;
+  time: string;
+  reason: string;
+  link: string;
+};
+
+function timeSlotMapper(t: TFunction, slot: Node): RejectedOccurrencesView {
+  const allocatedSlot = slot.recurringReservation?.allocatedTimeSlot;
+  const allocatedReservationUnit =
+    allocatedSlot?.reservationUnitOption.reservationUnit;
+  const allocatedReservationUnitName = allocatedReservationUnit?.nameFi ?? "-";
+  const allocatedUnit = allocatedReservationUnit?.unit?.nameFi ?? "-";
+
+  const application =
+    allocatedSlot?.reservationUnitOption.applicationSection?.application;
+  const applicantName = getApplicantName(application);
+
+  const day = allocatedSlot?.dayOfTheWeek
+    ? convertWeekday(allocatedSlot?.dayOfTheWeek)
+    : 0;
+  const begin = allocatedSlot?.beginTime ?? "";
+  const end = allocatedSlot?.endTime ?? "";
+  const timeString = `${t(`dayShort.${day}`)} ${begin.slice(0, 5)} - ${end.slice(0, 5)}`;
+  const name =
+    allocatedSlot?.reservationUnitOption.applicationSection.name ?? "-";
+
+  const applicationPk = application?.pk ?? 0;
+  const reservationPk = slot.recurringReservation?.reservations[0]?.pk ?? null;
+  const link = getReservationUrl(reservationPk);
+  const reason = slot.rejectionReason ?? "-";
```

Comment:
This doesn't require translation?

Reply:
I'd imagine it might, but the ticket doesn't specify any translations as is.